trin-execution/src/evm/pre_block_contracts.rs

```diff
@@ -0,0 +1,59 @@
+use alloy_eips::eip4788::{self, BEACON_ROOTS_ADDRESS};
+use anyhow::anyhow;
+use ethportal_api::Header;
+use revm::{db::State, DatabaseCommit, Evm};
+use revm_primitives::{SpecId, TxEnv, TxKind, U256};
+use trin_evm::spec_id::get_spec_id;
+
+use crate::storage::evm_db::EvmDB;
+
+/// Apply the beacon roots contract from eip-4788
+fn apply_beacon_root_contract(
+    evm: &mut Evm<(), State<EvmDB>>,
+    header: &Header,
+) -> anyhow::Result<()> {
+    if !get_spec_id(header.number).is_enabled_in(SpecId::CANCUN) {
+        return Ok(());
+    }
+
+    let Some(parent_beacon_block_root) = header.parent_beacon_block_root else {
+        return Err(anyhow!("Parent beacon block root is missing"));
+    };
+
+    // Save the previous environment
+    let previous_env = Box::new(evm.context.evm.env().clone());
+
+    // Update transaction environment to call the beacon roots contract
+    evm.context.evm.env.tx = TxEnv {
+        caller: eip4788::SYSTEM_ADDRESS,
+        transact_to: TxKind::Call(BEACON_ROOTS_ADDRESS),
```

Comment:
```suggestion
        transact_to: TxKind::Call(eip4788::BEACON_ROOTS_ADDRESS),
```

nit: for consistency with previous line